src/simtools/simtel/simtel_io_event_writer.py

```diff
@@ -15,144 +18,265 @@
     TriggerInformation,
 )
 
+from simtools.corsika.primary_particle import PrimaryParticle
+from simtools.simtel.simtel_io_file_info import get_corsika_run_header
 from simtools.utils.geometry import calculate_circular_mean
 
-DEFAULT_FILTERS = tables.Filters(complevel=5, complib="zlib", shuffle=True, bitshuffle=False)
-
 
 @dataclass
-class ShowerEventData:
-    """Shower event data."""
-
-    simulated_energy: list = field(default_factory=list)
-    x_core: list = field(default_factory=list)
-    y_core: list = field(default_factory=list)
-    shower_azimuth: list = field(default_factory=list)
-    shower_altitude: list = field(default_factory=list)
-    shower_id: list = field(default_factory=list)
-    area_weight: list = field(default_factory=list)
-
-    x_core_shower: list = field(default_factory=list)
-    y_core_shower: list = field(default_factory=list)
-    core_distance_shower: list = field(default_factory=list)
-
-
-@dataclass
-class TriggeredEventData:
-    """Triggered event data."""
-
-    triggered_id: list = field(default_factory=list)
-    array_altitudes: list = field(default_factory=list)
-    array_azimuths: list = field(default_factory=list)
-    trigger_telescope_list_list: list = field(default_factory=list)
-    angular_distance: list = field(default_factory=list)
+class TableSchemas:
+    """Define schemas for output tables with units."""
+
+    shower_schema = {
+        "shower_id": (np.uint32, None),
+        "event_id": (np.uint32, None),
+        "file_id": (np.uint32, None),
+        "simulated_energy": (float, u.TeV),
+        "x_core": (float, u.m),
```

Comment:
Could use also np floats here to be consistent.

Reply:
Changed to np.float64